server/server.go

```diff
@@ -2042,6 +2042,16 @@ func (s *Server) StartMonitoring() error {
 	// Snapshot server options.
 	opts := s.getOpts()
 
+	// Used to track HTTP requests
+	s.httpReqStats = map[string]uint64{
```

Comment:
I am confused. StartMonitoring() is calling s.startMonitoring() where this initialization was made. Can you explain to me what different this makes?

Reply:
`s.startMonitoring` is only called if an http or https port is configured.

Reply:
So you don't define an HTTP port and you still register the public handlers. So now you want to do the init here so that you call StartMonitoring(), knowing that it won't call startMonitoring(), right?

If I understood correctly, then I am thinking that we could initialize this map when creating the server instance. That would be the common place and ok I guess, even if one does not use monitoring.

Reply:
That is the behavior we have in our Fork, I am happy to adjust it that way, just the constructor for server was pretty busy already, but makes sense from an API standpoint.

Reply:
A simple:
```
s := &Server{
...
    eventIds:     nuid.New(),
    routesToSelf: make(map[string]struct{}),
    httpReqStats: make(map[string]uint64), // Used to track HTTP requests
}
```
would do I think...

Reply:
This wouldn't cover the use cases of Starting a server that wasn't called through New, is that ok?

Reply:
Updated to constructor init.

Reply:
> This wouldn't cover the use cases of Starting a server that wasn't called through New, is that ok?

NewServer() is called in all situations. Even New() calls NewServer() but simply ignore the error (legacy API).